objective.py

```diff
@@ -5,70 +5,86 @@
 
-    from skglm.datafits import QuadraticGroup
-    from skglm.penalties import WeightedGroupL2
     from skglm.utils.data import grp_converter
 
 
 class Objective(BaseObjective):
-    name = "Group Lasso objective"
-
-    requirements = ["pip:skglm"]
-
+    name = "Sparse Group Lasso"
+
+# skglm is needed here to create group partition and indices
+    requirements = [
+        "numpy'<2'",
+        "pip::git+https://github.com/scikit-learn-contrib/skglm",
+    ]
+    # tau = 0 is Group Lasso
     parameters = {
-        'reg': [1., 1e-1, 1e-2, 1e-3],
+        'tau': [0, 0.5, 0.9],
+        'reg': [1., 1e-1, 1e-2]
     }
 
-    def __init__(self, reg=1e-1):
-        self.reg = reg
+    def value_penalty(self, w):
+        """Value of penalty at vector ``w``."""
 
-    def set_data(self, X, y, groups):
-        _, n_features = X.shape
+        sum_penalty = 0.
+        for g in range(self.n_groups):
+            grp_g_indices = self.grp_indices[
+                self.grp_ptr[g]: self.grp_ptr[g+1]]
+            w_g = w[grp_g_indices]
 
-        grp_indices, grp_ptr = grp_converter(groups, n_features)
-        n_groups = len(grp_ptr) - 1
-        weights = np.ones(n_groups)
+            sum_penalty += norm(w_g)
+        sum_penalty *= (1 - self.tau)
+        sum_penalty += self.tau * np.sum(np.abs(w))
 
-        lmbd_max = self._compute_lmbd_max(X, y, grp_indices, grp_ptr)
-        self.lmbd = self.reg * lmbd_max
+        return self.lmbd * sum_penalty
+
+    def set_data(self, X, y, groups):
+        self.n_samples, self.n_features = X.shape
+
+        self.grp_indices, self.grp_ptr = grp_converter(groups, self.n_features)
+        self.n_groups = len(self.grp_ptr) - 1
 
         self.X, self.y = X, y
         self.groups = groups
-        self.grp_indices, self.grp_ptr = grp_indices, grp_ptr
 
-        self.datafit = QuadraticGroup(grp_ptr, grp_indices)
-        self.penalty = WeightedGroupL2(
-            self.lmbd, weights, grp_ptr, grp_indices
-        )
+        lmbd_max = self._compute_lmbd_max()
+        self.lmbd = self.reg * lmbd_max
 
-    def evaluate_result(self, w):
-        X, y = self.X, self.y
-        datafit, penalty = self.datafit, self.penalty
+    def evaluate_result(self, beta):
+        diff = self.y - self.X @ beta
+        datafit = .5 * diff @ diff / self.n_samples
+        pen = self.value_penalty(beta)
 
         return dict(
-            value=datafit.value(y, w, X @ w) + penalty.value(w)
+            value=datafit+pen,
         )
 
     def get_objective(self):
         return dict(
             X=self.X, y=self.y, lmbd=self.lmbd, groups=self.groups,
-            grp_indices=self.grp_indices, grp_ptr=self.grp_ptr
+            grp_indices=self.grp_indices, grp_ptr=self.grp_ptr, tau=self.tau
         )
 
-    def _compute_lmbd_max(self, X, y, grp_indices, grp_ptr):
-        lmbd_max = 0.
-        n_samples = X.shape[0]
-        n_groups = len(grp_ptr) - 1
+    def _compute_lmbd_max(self):
```

Comment:
As pointed out by @mathurinm in https://github.com/Badr-MOUFAD/benchmark_group_lasso/pull/1/files#r1701936766,
this is a heuristic to compute the lambda max (when ``tau == 0``, it is provides the write `lmbd_max`

We better add a comment to mention that